service/src/main/java/com/ge/predix/acs/attribute/readers/ExternalAttributeReader.java

```diff
@@ -92,20 +103,36 @@ private OAuth2RestTemplate getAdapterOauth2RestTemplate(final AttributeAdapterCo
             String adapterUrl = UriComponentsBuilder.fromUriString(adapterConnection.getAdapterEndpoint())
                     .queryParam(ID, identifier).toUriString();
 
+            AttributesResponse attributesResponse;
+
             try {
-                ResponseEntity<AttributesResponse> attributesResponse = adapter.getForEntity(adapterUrl,
-                        AttributesResponse.class);
-                attributes = attributesResponse.getBody().getAttributes();
+                attributesResponse = adapter.getForEntity(adapterUrl, AttributesResponse.class).getBody();
             } catch (final Throwable e) {
-                throw new AttributeRetrievalException(AttributeRetrievalException.getAdapterErrorMessage(adapterUrl),
-                        e);
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug(AttributeRetrievalException.getAdapterErrorMessage(adapterUrl), e);
+                }
+                attributesWrapper.setException(
```

Comment:
don't think ACS should treat this  the same as a response size being too large.  This needs to be handled with a circuit breaker (separate story). For now, it should just retry on the next request ?